src/tennissimulator/match/Match.java

```diff
@@ -0,0 +1,84 @@
+package tennissimulator.match;
+
+import tennissimulator.player.Player;
+
+import java.util.Optional;
+
+public class Match {
+
+    private final Player firstPlayer;
+    private final Player secondPlayer;
+    private ScoreTable scoreTable;
+    public static final int SETS = 3;
+    private Set set;
+    public static final int POINTS_IN_SET_TIEBREAK = 7;
+    public static final int POINTS_IN_MATCH_TIEBREAK = 10;
+    private int firstPlayerSetsWon;
+    private int secondPlayerSetsWon;
+
+    public Match(Player firstPlayer, Player secondPlayer) {
+        this.firstPlayer = firstPlayer;
+        this.secondPlayer = secondPlayer;
+        initialize();
+    }
+
+    private void initialize() {
+        scoreTable = new ScoreTable(firstPlayer, secondPlayer);
+        set = new Set(firstPlayer, secondPlayer, scoreTable);
+        firstPlayerSetsWon = 0;
+        secondPlayerSetsWon = 0;
+    }
+
+    public void simulate() {
+        defineFirstServer();
+        while (hasNoWinner()) {
+            set.simulate();
```

Comment:
Why this set.simulate() is needed? You don't even check who won

Reply:
typing error, I forget to remove it